src/go/configinfo/service_info.go

```diff
@@ -321,7 +321,11 @@ func addHttpRule(method *methodInfo, r *annotationspb.HttpRule, httpPathWithOpti
 			UriTemplate: r.GetCustom().GetPath(),
 			HttpMethod:  r.GetCustom().GetKind(),
 		}
-		httpPathWithOptionsSet[r.GetCustom().GetPath()] = true
+		matcher, err := util.GetRegexForUriWithPathParams(r.GetCustom().GetPath())
```

Comment:
I don't understand here.  Why we only call GetRegex for "custom" type,  not for other methods?

Reply:
I see.  the code assume Custom is OPTIONS.  actually,  custom can be any method.   We should check
if r.GetCustom().GetKind() == OPTIONS

Reply:
Good point, done. I assume custom options will only be used for OPTIONS, but we can double check to be sure.